src/tribler-gui/tribler_gui/core_manager.py

```diff
@@ -87,20 +89,65 @@ def start(self, core_args=None, core_env=None):
         First test whether we already have a Tribler process listening on port <CORE_API_PORT>.
         If so, use that one and don't start a new, fresh session.
         """
+        root_state_dir = get_root_state_directory()
 
         def on_request_error(_):
             self.use_existing_core = False
             self.start_tribler_core(core_args=core_args, core_env=core_env)
 
+        do_cleanup, old_version_dirs = self.should_cleanup_old_versions(root_state_dir, version_id)
+        if do_cleanup:
+            for version_dir in old_version_dirs:
+                shutil.rmtree(str(version_dir), ignore_errors=True)
+
+        # Connect to the events manager only after the cleanup is done
         self.events_manager.connect()
```

Comment:
Trying to connect the core after the cleanup is done should address the issue about GUI being stuck if user waits long. There is no need to increase the timeout or pause the connection attempts with this.